lib/src/wiggle_abi/cache.rs

```diff
@@ -223,15 +238,41 @@ impl FastlyCache for Session {
         options_mask: types::CacheLookupOptionsMask,
         options: wiggle::GuestPtr<types::CacheLookupOptions>,
     ) -> Result<types::CacheBusyHandle, Error> {
-        Err(Error::NotAvailable("Cache API primitives"))
+        let headers = load_lookup_options(self, memory, options_mask, options)?;
+        let key = load_cache_key(memory, cache_key)?;
+        let cache = Arc::clone(self.cache());
+
+        // Look up once, joining the transaction only if obligated:
+        let e = cache.transaction_lookup(&key, &headers, false).await;
+        let ready = e.found().is_some() || e.go_get().is_some();
+        // If we already got _something_, we can provide an already-complete PeekableTask.
+        // Otherwise we need to spawn it and let it block in the background.
+        let task = if ready {
+            PeekableTask::complete(e)
+        } else {
+            PeekableTask::spawn(Box::pin(async move {
+                Ok(cache.transaction_lookup(&key, &headers, true).await)
```

Comment:
The double lookup here is slightly unfortunate, but ultimately harmless -- and treating this "atomically" would require a bit more contortion, so, 👍